src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexParser.cs

```diff
@@ -2081,6 +2076,27 @@ private bool IsCaptureSlot(int i)
             // '  a  b  c  d  e  f  g  h  i  j  k  l  m  n  o  p  q  r  s  t  u  v  w  x  y  z  {  |  }  ~
                0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, Q, S, 0, 0, 0};
 
+#if NET7_0_OR_GREATER
```

Comment:
I merged the PR, but... this should be NET8_0_OR_GREATER.  Seems like an artifact of our build system right now that this work.  Once we fix that, these are going to start to fail presumably.
cc: @ViktorHofer

Reply:
Right, the `NET8` directives don't seem to exist yet.

Once they are added, would we ever build this with an actual 7.0 target? That is, would this continue to just "work", while being misleading to the reader since the API doesn't actually exist on 7?

Reply:
It might.  We have several libraries that build for older .NET Core versions, and while System.Text.RegularExpressions.dll itself isn't one of them, it's feasible the source generator could.

Plus, it's confusing.

Reply:
Personally, I would have used `#if NET` here instead as presumably at the time that we would ship a .NETCoreApp version of the Regex source generator (which isn't planned), we wouldn't build older tfms like net6.0 and net7.0 anymore.

Can you please either submit a PR to change this to `#NET` or create an issue that tracks updating this to `NET8_0_OR_GREATER`?

Reply:
> Personally, I would have used #if NET here

FWIW I think that's just as if not more confusing. All of this is .NET.

Reply:
Then you can use `#if NETCOREAPP` which means the same thing as `#if NET`

Reply:
Same objection.

Reply:
Both .NET 7 and .NET 8 are netcoreapp.

Reply:
That's why I suggested to use `#if NET` as if we would ever make the source generator target `NETCOREAPP` we likely wouldn't build `net7.0` anymore. We don't version preprocessor directives in the BCL if there isn't a current need. I.e., there aren't any `NETCOREAPP_3_1_OR_GREATER` (or earlier) symbols because we don't build such frameworks anymore.

Reply:
.NET Core 3.1 is old and unsupported. We just shipped NET 7 last week. I'm glad it's not confusing to you; it's highly confusing to me.